### ISSUE TITLE
Error deploying Bicep: Failed to parse template

### ISSUE THREAD

**Issue by ericscheffler** [Azure/azure-cli#25890]
## Describe the bug
Receiving errors when trying to deploy a known-good and tested Bicep template using 'az deployment group create' or 'az deployment group what-if'.
Error: "Failed to parse '.\path\to\main.bicep', please check whether it is a valid JSON format"
Same template works on an older machine, and other colleagues are noting the same behavior. "az bicep build" runs normally and compiles a valid ARM template (which deploys successfully so I know the template works), and deploying using Powershell below works fine.
```
$params = @{
    ResourceGroupName     = 'rgname'
    TemplateFile          = '.\path\to\main.bicep'
    TemplateParameterFile = '.\path\to\params.json'
}

Test-AzResourceGroupDeployment @params -Verbose
New-AzResourceGroupDeployment @params -Verbose
```

**Command Name**
`az deployment group create`
`az deployment group what-if`

**Errors:**
```
cli.azure.cli.core.azclierror: Failed to parse '.\path\to\main.bicep', please check whether it is a valid JSON format
az_command_data_logger: Failed to parse '.\path\to\main.bicep', please check whether it is a valid JSON format
```
*** See attached .txt file for full debug output

## To Reproduce:
Deploy any working Bicep template using Az CLI:
$rg= "resourcegroupname"
$template= ".\Path\To\pipprefix.bicep"
$params= ".\Path\To\params.json"

az deployment group create --resource-group $rg
--template-file $template `
--parameters $params

## Expected Behavior
Template should deploy a VWAN instance with 3 secured hubs and an AZFW policy.

## Environment Summary
```
Windows-10-10.0.22621-SP0
Python 3.10.10
Installer: MSI

azure-cli 2.46.0
```
## Additional Context

[DebugInfo.txt](https://github.com/Azure/azure-cli/files/11022866/DebugInfo.txt)

**Comment by yonzhan** [comment]
route to CXP team

**Comment by ericscheffler** [comment]
I've updated to Bicep CLI 0.15.31 and az cli 2.47 with no change, deployments still don't work with the error noted above.

**Comment by shenglol** [comment]
@ericscheffler I came across this today and did some investigation. Turns out it is the Bicep tracing message (`TRACE:Building semantic model for file`) that messes up the JSON output, which should be a Bicep bug. I've created https://github.com/Azure/bicep/issues/10477 to track it.

The workaround is to set the `BICEP_TRACING_ENABLED` environment variable to "False".

@yonzhan We may close this since it's not an Azure CLI bug.
